Sources/Native/NativeMockServerWrapper.swift

```diff
@@ -2,102 +2,206 @@ import Foundation
 import SwiftyJSON
 import BrightFutures
 
-class NativeMockServerWrapper: MockServer {
-  var port: Int32 = -1
-  var pactDir: String = ""
-
-  public init() {
-    pactDir = defaultPactDir()
-    port = randomPort()
-  }
-
-  func defaultPactDir() -> String {
-    if let dir = ProcessInfo.processInfo.environment["pact_dir"] {
-      return dir
+#if SWIFT_PACKAGE
+    import SwiftPactServer
+#endif
+
+public class NativeMockServerWrapper: MockServer {
+    private lazy var port: Int32 = {
+        return findUnusedPort()
+    }()
+    private let pactDir: String = ProcessInfo.processInfo.environment["pact_dir"] ?? "/tmp/pacts"
+    private let shouldWritePacts: Bool
+
+    public init(shouldWritePacts: Bool = false) {
```

Comment:
is `shouldWritePacts` used to have a dry run on tests when it's set to `false`?

Reply:
If I remember it correctly this was needed for the setup I'm using.
I think part of the problem was this is failing on devices and it only worked for simulators. So I had to control a bit when to write for being able to automate this in our building pipeline.